plugin/pkg/admission/priority/admission.go

```diff
@@ -162,6 +163,15 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 				return fmt.Errorf("failed to get default priority class: %v", err)
 			}
 		} else {
+			pcName := pod.Spec.PriorityClassName
+			// Only allow system priorities in the system namespace. This is to prevent abuse or incorrect
+			// usage of these priorities. Pods created at these priorities could preempt system critical
+			// components.
+			for _, spc := range scheduling.SystemPriorityClasses() {
+				if spc.Name == pcName && a.GetNamespace() != metav1.NamespaceSystem {
```

Comment:
Will default priority be the SystemCritical ? or let cluster admin to decide that.

Reply:
should we also check it for `ExperimentalCriticalPodAnnotation` ?

Reply:
Default priority is a flag on the PriorityClass that can be set by an admin.
Checking ExperimentalCriticalPodAnnotation is not needed here.

Reply:
> should we also check it for ExperimentalCriticalPodAnnotation ?

that's already restricted to the kube-system namespace, so no need to do anything additional here